### PR TITLE
fix(totp): add totp code window valid config

### PR DESCRIPTION
**Pull request by vbudhram**
This PR adds a configurable window for TOTP codes to be valid, defaulting to 1.

Fixes https://github.com/mozilla/fxa-content-server/issues/6005

### CODE REVIEW

File: test/remote/totp_tests.js [hunk 1]
```diff
@@ -206,6 +208,21 @@ describe('remote totp', function () {
           assert.equal(emailData.headers['x-template-name'], 'newDeviceLoginEmail', 'correct template sent')
         })
     })
+
+    it('should verify totp one outside one code window', () => {
+      const code = otplib.authenticator.generate()
+      return P.delay(4000)
```
Comment (vbudhram) [inline test/remote/totp_tests.js]: Looking through previous remote tests, each consistently finishing in under 2 seconds, so having a 4 second delay here seemed like a safe choice for a code window.

**Comment by vbudhram** [comment]
Test failures fixed with https://github.com/mozilla/fxa-auth-db-mysql/pull/330

**Comment by vbudhram** [comment]
@mozilla/fxa-devs r?

File: config/index.js [hunk 2]
```diff
@@ -797,6 +797,12 @@ var conf = convict({
       format: 'nat',
       env: 'TOTP_STEP_SIZE'
     },
+    window: {
+      doc: 'Tokens in the previous x-windows that should be considered valid',
+      default: 1,
+      format: 'nat',
+      env: 'TOTP_CODE_WINDOw'
```
Comment (vladikoff) [inline config/index.js]: typo `TOTP_CODE_WINDOw`

Comment (vladikoff) [reply config/index.js]: nit: can we name this `totpWindow` ? 

1. env var is called `..._CODE_WINDOW`
(2. and also `window` is a browser global which is a bit trippy)

File: lib/routes/totp.js [hunk 3]
```diff
@@ -18,7 +18,8 @@ module.exports = (log, db, mailer, customs, config) => {
   // Default options for TOTP
   otplib.authenticator.options = {
     encoding: 'hex',
-    step: config.step
+    step: config.step,
+    window: config.window
```
Comment (vladikoff) [inline lib/routes/totp.js]: if the nit is `meh` above, I think you can also do 

```js
endoing: 'hex', 
step,
window
```

File: test/remote/totp_tests.js (hunk 1 above)
Comment (vladikoff) [reply test/remote/totp_tests.js]: we cannot wait for a test to be done in 30 secs, even 4 secs is very long time for this test. Can we avoid the 4 secs and test some other way?

Comment (gdestuynder) [reply test/remote/totp_tests.js]: Generate TOTPs that are 20s ahead and 20s behind window (to check the window is ~90s) and/or 31s ahead/31s behind (to check it fails if outside 90s window), then verify them - should be faster than the 4s delay 

At least, that's what I would do - I don't know if it makes sense for you or not

File: lib/routes/totp.js [hunk 4]
```diff
@@ -64,7 +65,9 @@ module.exports = (log, db, mailer, customs, config) => {
           .then(createResponse)
           .then(() => reply(response), reply)
 
-        const secret = otplib.authenticator.generateSecret()
+        const authenticator = new otplib.authenticator.Authenticator()
```
Comment (vbudhram) [inline lib/routes/totp.js]: Not directly related to code window size, but changed this to be consistent with how we use `otplib.authenticator` in other files and repos.

File: test/local/routes/totp.js [hunk 5]
```diff
@@ -7,12 +7,13 @@
 const assert = require('insist')
 const getRoute = require('../../routes_helpers').getRoute
 const mocks = require('../../mocks')
+const otplib = require('otplib')
 const P = require('../../../lib/promise')
 const sinon = require('sinon')
-const proxyquire = require('proxyquire').noPreserveCache()
```
Comment (vbudhram) [inline test/local/routes/totp.js]: Benefit from https://github.com/mozilla/fxa-auth-server/pull/2371/files#r178127428, is that we no longer need proxyrequire.

File: test/remote/totp_tests.js (hunk 1 above)
Comment (vbudhram) [reply test/remote/totp_tests.js]: Updated this to remove the delay and check against future and previous code windows.

**Pull request merged**